### PR TITLE
Count time between pulses instead of pulses in a period.

### PR DESCRIPTION
**Pull request by joaoantoniocardoso**

### CODE REVIEW

File: firmware/src/machine.c [hunk 1]
```diff
@@ -352,10 +339,31 @@ ISR(TIMER2_COMPA_vect)
...
+    static uint16_t t1 = 0;
+    static uint16_t t2 = 0;
+
+    if(tachometer.started == 0){
+        t1 = time();
+        tachometer.started = 1;
+    }else{
+        t2 = time();
+
+        tachometer.rpm_avg_sum += ((uint32_t)(t2 - t1))
```
Comment (joaoantoniocardoso) [inline firmware/src/machine.c]: This is actually the accumulation of delta-time, not RPM.

File: firmware/src/machine.c [hunk 2]
```diff
@@ -352,10 +339,31 @@ ISR(TIMER2_COMPA_vect)
...
+        tachometer.overflow_counter = 0;
+
+        t1 = t2;
     }
 }
 
+ISR(TIMER1_OVF_vect)
+{
+    if (++tachometer.overflow_counter == 0) {
+        tachometer.tachometer = 0;
```
Comment (AlejoPm03) [inline firmware/src/machine.c]:
```suggestion
        tachometer.started = 0;
```

File: firmware/src/can_app.c [hunk 3]
```diff
@@ -1,16 +1,29 @@
 #include "can_app.h"
 
+uint32_t can_app_send_state_clk_div;
+uint32_t can_app_send_tachometer_clk_div;
+
+extern tachometer_t tachometer;
+extern state_machine_t state_machine;
+extern error_flags_t error_flags;
```
Comment (joaoantoniocardoso) [inline firmware/src/can_app.c:8]: Could you move it to `machine.h`?

Comment (joaoantoniocardoso) [reply firmware/src/can_app.c:8]: Also, `extern` -> `extern volatile`.

File: firmware/src/can_app.c [hunk 4]
```diff
@@ -1,16 +1,31 @@
...
-        tachometer.rpm_avg_sum = 0;
-        tachometer.rpm_avg_sum_count = 0;
+    if(tachometer.dt_avg_sum_count){
+		tachometer.lock = 1;
+        tachometer.dt_avg = tachometer.dt_avg_sum/tachometer.dt_avg_sum_count;
     }
+    tachometer.dt_avg_sum = 0;
+    tachometer.dt_avg_sum_count = 0;
+	tachometer.lock = 0;
+	tachometer.rpm = 60 * 1e6 /  (tachometer.dt_avg * 64);
```
Comment (joaoantoniocardoso) [inline firmware/src/can_app.c:23]:
```suggestion
    static uint8_t zeroes_counter = 0;
    tachometer.lock = 1;
    if(tachometer.dt_avg_sum_count && tachometer.dt_avg_sum){
        tachometer.dt_avg = tachometer.dt_avg_sum/tachometer.dt_avg_sum_count;
        tachometer.dt_avg_sum = 0;
        tachometer.dt_avg_sum_count = 0;
	      
	      tachometer.rpm = 60 * 1e6 / (tachometer.dt_avg * 64);
	      zeroes_counter = 0;
	  }else{
	      if(++zeroes_counter == 5) tachometer.rpm = 0;
	  }
	  tachometer.lock = 0;
```

reasoning:
- set RPM to zero after 5 average computation tries with incomplete turns.

note:
- tachometer.lock needs to be set before any read to tachometer variables

**Pull request merged**